### PR TITLE
Flow Otelcol: Document default values for gRPC client balancer_name

### PR DESCRIPTION
**Pull request by ptodev**
`balancer_name` has a default value which is not listed in the tables in our docs.

This was noticed by @mattdurham in [another PR](https://github.com/grafana/agent/pull/4148#discussion_r1237634250).

### CODE REVIEW

File: docs/sources/shared/flow/reference/components/otelcol-grpc-balancer-name.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+---
+aliases:
+- /docs/agent/shared/flow/reference/components/otelcol-grpc-balancer-name
+headless: true
+---
+
+The supported values for `balancer_name` are listed in the gRPC documentation on [Load balancing][]:
+* `pick_first`: Tries to connect to the first address, uses it for all RPCs if it connects, 
+  or tries the next address if it fails (and keep doing that until one connection is successful).
```
Comment (clayton-cornell) [inline docs/sources/shared/flow/reference/components/otelcol-grpc-balancer-name.md:9]:
```suggestion
  or tries the next address if it fails (and keeps doing that until one connection is successful). 
```

File: docs/sources/shared/flow/reference/components/otelcol-grpc-balancer-name.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+headless: true
+---
+
+The supported values for `balancer_name` are listed in the gRPC documentation on [Load balancing][]:
+* `pick_first`: Tries to connect to the first address, uses it for all RPCs if it connects, 
+  or tries the next address if it fails (and keep doing that until one connection is successful). 
+  Because of this, all the RPCs will be sent to the same backend.
+* `round_robin`: Connects to all the addresses it sees, and sends an RPC to each backend one at a time in order. 
+  E.g. the first RPC will be sent to backend-1, the second RPC will be be sent to backend-2, 
+  and the third RPC will be be sent to backend-1 again.
```
Comment (clayton-cornell) [inline docs/sources/shared/flow/reference/components/otelcol-grpc-balancer-name.md:13]:
```suggestion
  For example, the first RPC is sent to backend-1, the second RPC is sent to backend-2, 
  and the third RPC is sent to backend-1.
```

**Review by clayton-cornell — COMMENTED** [review]
A couple of suggestions for the docs

**Review by mattdurham — APPROVED** [review]
LGTM

**Review by clayton-cornell — APPROVED** [review]
Looks OK from a docs standpoint. I don't see any issues that stand out

**Pull request merged**